bionic/flow.py

```diff
@@ -927,9 +948,11 @@ def __call__(self, func):
             {len(docs)} docs {tuple(docs)!r}"""
             raise ValueError(oneline(message))
 
+        internal_names = [name for name in names if entity_is_internal(name)]
         can_persist = acc.can_persist
         if can_persist is None:
-            can_persist = True
+            can_persist = len(internal_names) == 0
```

Comment:
Someone can technically redefine internal entities using a function. Something like:
```
    @builder
    def core__persistent_cache__global_dir():
        return "mycachedir"
```

I don't see this happening in practice but it's better to take care of it now. 

Maybe the right answer here is that we stop redefining entities (or wiping current state) when setting entities using a function. That's one of the proposals in issue #115.

Reply:
They can, but it will throw an exception and tell them to use `@immediate`, right? Isn't that the current behavior as well (but with the exception coming from a different place)? Since this is a fairly rare case and the ramifications may not be obvious to the user, it feels better to me to give them a detailed warning and force them to be explicit.

BTW, I'm in favor of not redefining entities in this case -- we just need to figure out what our backwards compatibility story is.

Reply:
Yes but we will now emit a warning instead of throwing an exception. I can get behind the reason that the warning is better than this special logic though, as you said, it forces them to be explicit.

Reply:
Oh wait, I don't know how I missed this but you are right, we are validating in the entity defn and this will throw an error. I'm going to change this like you suggested.